ballerina/modules/parser/parser.bal

```diff
@@ -205,18 +252,24 @@ public class Parser {
         while token.kind != T_CLOSE_PARENTHESES {
             token = check self.readNextNonSeparatorToken();
             ArgumentName name = check getArgumentName(token);
-
             token = check self.readNextNonSeparatorToken();
             if token.kind != T_COLON {
                 return getExpectedCharError(token, COLON);
             }
-
             token = check self.readNextNonSeparatorToken();
-            ArgumentValue value = check getArgumentValue(token);
-
-            ArgumentNode argument = new(name, value, <ArgumentType>token.kind);
-            fieldNode.addArgument(argument);
-            token = check self.peekNextNonSeparatorToken();
+            if (token.kind == T_DOLLAR) {
+                token = check self.readNextNonSeparatorToken();
+                //send the variable name as argument value of the argument node
+                ArgumentValue varName = check getArgumentValue(token); 
+                ArgumentNode argument = new(name, varName, <ArgumentType>token.kind, true);
```

Comment:
since ```token.kind``` is checked at ```getArgumentValue()```,  we can guarantee that ```token.kind``` is always an ```ArgumentType```.  But i think, it's better to use local variable here.